patches/net/minecraft/world/entity/player/Player.java.patch

```diff
@@ -98,20 +98,26 @@
          float f = this.inventory.getDestroySpeed(p_36282_);
          if (f > 1.0F) {
              int i = EnchantmentHelper.getBlockEfficiency(this);
-@@ -740,11 +_,12 @@
+@@ -740,13 +_,19 @@
              f /= 5.0F;
          }
  
 +        f = net.neoforged.neoforge.event.EventHooks.getBreakSpeed(this, p_36282_, f, pos);
          return f;
      }
  
++    @Deprecated // Neo: use position-sensitive version below
      public boolean hasCorrectToolForDrops(BlockState p_36299_) {
```

Comment:
Wouldn't it make more sense to make the other method accept `nullable` parameters and forward to it?

Reply:
As the call chain basically goes new method -> event firing -> old method if event didn't yield a result, I don't think that's a good idea, no.

Reply:
I am making that suggestion exactly because I don't like that call chain. It's confusing for the deprecated method to still be called but later in the chain, IMO.